### DISCUSSION TITLE
zephyr+Linux+hypervisor on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by wjhtinger** [zephyrproject-rtos/zephyr#43427] [Q&A]
I want to run zephyr+Linux on Raspberry Pi 4 based on a hypervisor, just zephyr use one core and Linux use left three core. 
But now zephyr do not support Raspberry Pi 4, right? So, how can I port the zephyr to Raspberry Pi 4? Anyone can give some ideals?

**Comment by dbaluta** [comment]
FIrst you will need to identify which ARM cores are used on Raspberry Pi 4. I remember that at some point there were some efforts to port Zephyr on ARM A cores. 

Also, which hypervisor are you thinking of?

**Comment by wjhtinger** [comment]
Raspberry Pi 4 CPU spec is : Broadcom BCM2711, Quad core Cortex-A72 (ARM v8) 64-bit SoC @ 1.5GHz. So, Dose Zephyr support it?
And in my hybrid system, Zephyr dont use devices except a uart, Linux will use most devices.

Now Raspberry Pi is so popular, why not add it to the Zephyr project?

As to hypervisor, I will use a very small and simple one, just like this: https://github.com/bao-project/bao-hypervisor

**Comment by stephanosio** [comment]
This might be a good read: https://github.com/zephyrproject-rtos/zephyr/issues/25182

In case of hypervisors, that would have more to do with adding the Zephyr port for such a hypervisor (e.g. jailhouse) than adding the support for Raspberry Pi 4 itself.

**Reply by wjhtinger** [reply]
So， what do you think should add to Zephyr for the  hypervisor?

> This might be a good read: #25182
> 
> In case of hypervisors, that would have more to do with adding the Zephyr port for such a hypervisor (e.g. jailhouse) than adding the support for Raspberry Pi 4 itself.

**Reply by stephanosio** [reply]
That really depends on which hypervisor you are targeting, how the hypervisor works, etc.
